src/install.c

```diff
@@ -343,19 +343,30 @@ static gboolean slot_list_contains(GList *slotlist, const RaucSlot *testslot)
 	return FALSE;
 }
 
-static RaucSlot* get_bootname_slot(GHashTable *targetgroup, GError **error)
+/* Get the slot that has a bootname set, i.e. that we need to manipulate by the
+ * boot selection layer to allow atomic updating.
+ */
+static RaucSlot* get_bootname_slot(GHashTable *targetgroup, GList *install_images, GError **error)
 {
-	GHashTableIter iter;
-	RaucSlot *iterslot = NULL;
 	RaucSlot *bootnameslot = NULL;
 
-	/* Sanity check: only one bootname allowed */
-	g_hash_table_iter_init(&iter, targetgroup);
-	while (g_hash_table_iter_next(&iter, NULL, (gpointer*) &iterslot)) {
+	/* Iterate over all slots we have an image to install for */
+	for (GList *l = install_images; l != NULL; l = l->next) {
+		RaucSlot *iterslot = g_hash_table_lookup(targetgroup, ((RaucImage*) l->data)->slotclass);
+
+		/* If our slot has a parent with bootname set, use this for
+		 * testing (setting both parent and bootname is not allowed) */
+		if (iterslot->parent)
+			iterslot = iterslot->parent;
 
 		if (!iterslot->bootname)
-			continue;
+				continue;
 
+		/* If we've detected a bootname slot while already having found
+		 * a bootname slot, this is an error and we must abort.
+		 * This can be caused either by system misconfiguration or by
+		 * cases where we have bad bundles with images for multiple
+		 * concurrent bootname slots. */
 		if (bootnameslot) {
```

Comment:
What if the bundle contains images for more than one slot of the targetgroup (e.g. _appfs_ and _rootfs_)? Shouldn't this be
```C
if (bootnameslot && bootnameslot != iterslot) {
```
instead?